scripts/setup_test.py

```diff
@@ -388,17 +439,18 @@ def mock_check_call(cmd_tokens):
                     common.YARN_VERSION, common.YARN_VERSION)])
         self.assertEqual(all_cmd_tokens, ['./configure', 'make'])
 
-    def test_package_install_with_windows_x86(self):
-        def mock_exists(unused_path):
+    def test_package_install_with_windows_x86(self) -> None:
+
+        def mock_exists(unused_path: str) -> Literal[False]:
             return False
 
-        def mock_url_retrieve(url, filename): # pylint: disable=unused-argument
+        def mock_url_retrieve(url: str, filename: str) -> None: # pylint: disable=unused-argument
             self.urls.append(url)
 
-        def mock_check_call(commands):
-            mock_check_call.commands = commands
+        def mock_check_call(commands: List[str]) -> None:
+            self.commands = commands
```

Comment:
If this is just to keep a record of what commands were used for later assertion, can you just use `self.swap_with_checks` instead? Alternatively, you could just make `self.commands` a local variable since AFAICT there's no reason for this to be an instance variable

Reply:
Okay. I will make it a local variable.

Reply:
I think we can't declare `self.commands` as a local variable because that will give errors like this
### Attribute 'commands' defined outside __init__
and that's why I initialized it within the setup function. should I keep it as it is?

Reply:
If you made `self.commands` a local variable, it would just be `commands`, not an attribute of `self`. Your code would look something like this:

```python
commands = []
...
def mock_check_call(commands_arg):
    commands = commands_arg
...
self.assertEqual(commands, ...)
```

However, I think `self.swap_with_checks` is a cleaner solution to using a local variable.